### PR TITLE
fixes #9073, #9251 - Add parser for facts 

### PR DESCRIPTION
**Pull request by stbenjam**

### CODE REVIEW

File: app/services/foreman_salt/fact_parser.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
+module ForemanSalt
+  class FactParser < ::PuppetFactParser
```
Comment (ohadlevy) [inline app/services/foreman_salt/fact_parser.rb]: does it really make sense to inherit from puppet? is there anything in the puppet parser that can be moved to the parent base class instead?

File: app/services/foreman_salt/fact_parser.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      ip[1] if ip
+    end
+
+    def primary_interface
+      interface = interfaces.find { |_, value| value[:ipaddress] == ip }
+      interface[0] if interface
+    end
+
+    def mac
+      interface = interfaces.find { |_, value| value[:ipaddress] == ip }
```
Comment (ohadlevy) [inline app/services/foreman_salt/fact_parser.rb]: mac or ip?

File: app/services/foreman_salt/fact_parser.rb (hunk 1 above)
Comment (stbenjam) [inline app/services/foreman_salt/fact_parser.rb]: How Salt handles operatingsystems is basically identical to puppet, and I didn't want to repeat all that logic.

But is it that useful to anybody else?  Chef or some other config mangaement system may use entirely different fact names and values.

**Comment by stbenjam** [comment]
Don't anyone merge yet, I need to check a debian system, I have a feeling Salt reports the release codename (e.g. wheezy) differently than Puppet facts.

File: app/services/foreman_salt/fact_parser.rb (hunk 1 above)
Comment (ares) [inline app/services/foreman_salt/fact_parser.rb]: so maybe we could extract common bits to some generic class and PuppetFactParser should inherit from it?

**Comment by stbenjam** [comment]
@domcleal Is it possible to kick off a jenkins build of this PR? It should fix http://projects.theforeman.org/issues/9251 too.

**Pull request merged**